build-tools/automation/azure-pipelines.yaml

```diff
@@ -240,6 +263,7 @@ stages:
         testRunTitle: Xamarin.Android.Build.Tests - Windows Build Tree
         testAssembly: $(System.DefaultWorkingDirectory)\bin\Test$(XA.Build.Configuration)\Xamarin.Android.Build.Tests.dll
         testResultsFile: TestResult-MSBuildTests-WinBuildTree-$(XA.Build.Configuration).xml
+        nunitConsoleExtraArgs: --workers=4
```

Comment:
Should this use `$(NUMBER_OF_PROCESSORS)` (assuming that you can access the `%NUMBER_OF_PROCESSORS%` environment variable...)?

Reply:
NUnitConsole will use all by default, but I wanted to limit this as I was seeing a consistent slowdown here. I had a hunch that we may be trying to do too much in parallel on these Windows machines, see https://github.com/xamarin/xamarin-android/pull/3860/commits/c5e291f709cd41901d58f1a1dc00f5f53059fec2.
Previous builds (such as https://dev.azure.com/devdiv/DevDiv/_build/results?buildId=3297546) hit this slowdown on the Windows build tree tests, and when this was happening we weren't able to remote into the machine at all...

Reply:
i.e. `%NUMBER_OF_PROCESSORS%` may in fact be too high, and unless YAML supports arithmetic -- which it doesn't appear to -- then hardcoding to a "known" lower value is the best we can do.

Right.  Thanks!

Reply:
Presumably YAML has a comment syntax; could we comment what we're trying to do here?

Reply:
Comment added in https://github.com/xamarin/xamarin-android/pull/3860/commits/7b38640bfb8895dc0a96ba5c28a34aae33e6b362.